src/main/java/org/jabref/gui/preferences/journals/JournalAbbreviationsTabViewModel.java

```diff
@@ -396,4 +403,24 @@ public SimpleBooleanProperty isAbbreviationEditableAndRemovable() {
     public SimpleBooleanProperty isFileRemovableProperty() {
         return isFileRemovable;
     }
+
+    public void filterAbbreviations(String text) {
+        if (text.isEmpty()) {
+            clearSearch();
+            return;
+        }
+
+        this.filteredAbbreviations.unbindBidirectional(this.abbreviations);
+        this.filteredAbbreviations.clear();
```

Comment:
I don't understand why this is not working.
Basically I keep two lists, one of all abbreviations and one of the search result. The search result is bound to the view.
When not performing a search, I want to display all abbreviations and therefore bind them together. When searching, I remove that binding, clear the search result (filteredAbbreviations) and iterate over all abbreviations and only add abbreviations that contain the search string.
What happens is that when I clear filteredAbbreviations, abbreviations is cleared as well and the list is empty.

Reply:
Isn’t this easier with a JavaFX FilteredList?

Reply:
Hi @k3KAW8Pnf7mkmdSMPHz27 
It's certainly easier using FilteredList, provided one knows about it's existence :) Thanks for the suggestion!
Using FilteredList works, but I'm not completely happy about how I update the view now. As far as I can tell, there is no property in FilteredList that one could bind to the TableView. Therefore I had to use listeners to update the view each time the list of abbreviations or the search term changes. Is there a better way I am not seeing?

Reply:
You simply call tableView.setItems(filteredList) https://code.makery.ch/blog/javafx-8-tableview-sorting-filtering/

For setting the predicate of the FilteredLIst you can also use some EasyBInd foo magic, see for example in PreferencesFilterDialog

Reply:
I am looking at this on a cellphone so perhaps I am missing something.

[`setItems`](https://openjfx.io/javadoc/16/javafx.controls/javafx/scene/control/TableView.html#setItems(javafx.collections.ObservableList)) should accept a FilteredList, and since it is an ObservableList, you shouldn't need to update it at all (the one call in setupbindings should be enough).

You'll probably need to bind to the `predicateProperty` of the list, instead of using the constructor with a predicate that I linked (sorry X) )

Reply:
Thanks for your suggestions.
I did not change the predicate setting to EasyBind, since I see no major benefit in readability or functionality, I hope that's ok.